source/hud/blocklyButton.js

```diff
@@ -22,8 +22,16 @@ this.onClick = function() {
     var hud = this.parent;
     hud.lastBlocklyNodeID = hud.lastBlocklyNodeID || this.scene.player.rover.id;
     if ( hud.lastBlocklyNodeID !== undefined ) {
+        
+        var blocklyNode = blocklyNodes[ hud.lastBlocklyNodeID ];
+
+        if ( blocklyNode && Blockly.mainWorkspace ) {
+            Blockly.mainWorkspace.maxBlocks = blocklyNode.ramMax;
+        } else {
+            console.log("Either blocklyNode or Blockly.mainWorkspace are missing!");
+        }
```

Comment:
It is undefined when the property that the node belongs to is not a blockly node. That's why it did nothing for you. `blockly_activeNodeID` is a property of the scene, so the `nodeID` is the app id. It looks like you already fixed it, though.